src/syscheckd/db/include/db.hpp

```diff
@@ -8,1073 +8,48 @@
- * @param w_evt Whodata information
- *
- * @return FIMDB_OK on success, FIMDB_ERR otherwise.
- */
-int fim_db_process_missing_registry_data_entry(fdb_t* fim_sql, fim_tmp_file* file, pthread_mutex_t* mutex, int storage,
-                                               fim_event_mode mode, whodata_evt* w_evt);
-
-
-/**
- * @brief Get count of all entries in registry key and registry data table.
- *
- * @param fim_sql FIM database struct.
- *
- * @return Number of entries in registry key table.
- */
-int fim_db_get_count_registry_key_data(fdb_t* fim_sql);
-
-/**
- * @brief Delete registry using registry entry.
- *
- * @param fim_sql FIM database struct.
- * @param entry Registry entry.
- */
-int fim_db_remove_registry_key(fdb_t* fim_sql, fim_entry* entry);
-
-/**
- * @brief Delete registry data using fim_registry_value_data entry.
- *
- * @param fim_sql FIM database struct.
- * @param entry fim_registry_value_data entry.
- *
- * @return FIMDB_OK on success, FIMDB_ERR otherwise.
- */
-int fim_db_remove_registry_value_data(fdb_t* fim_sql, fim_registry_value_data* entry);
-
-/**
- * @brief Get a registry using it's id.
- *
- * @param fim_sql FIM database struct.
- * @param id Id of the registry key
- *
- * @return fim_registry_key structure.
- */
-fim_registry_key* fim_db_get_registry_key_using_id(fdb_t* fim_sql, unsigned int id);
-
-/**
- * @brief Get all registry values from given id.
- *
- * Given an id, save in a fim_tmp_file all its values.
- *
- * @param fim_sql FIM database struct.
- * @param file Structure of the file which contains all the paths.
- * @param storage Type of storage (memory or disk).
- * @param key_id Key id of the values.
- *
- * @return FIMDB_OK on success, FIMDB_ERR otherwise.
- */
-int fim_db_get_values_from_registry_key(fdb_t* fim_sql, fim_tmp_file** file, int storage, unsigned long int key_id);
-
-/**
- * @brief Decodes a row from the database to be saved in a fim_registry_key structure.
- *
- * @param stmt The statement to be decoded.
- *
- * @return fim_registry_key* The filled structure.
- */
-fim_registry_key* fim_db_decode_registry_key(sqlite3_stmt* stmt);
-
+#ifndef WIN32
 /**
- * @brief Decodes a row from the database to be saved in a fim_registry_value_data structure.
- *
- * @param stmt The statement to be decoded.
+ * @brief Initialize the FIM database.
  *
- * @return fim_registry_value_data* The filled structure.
+ * It will be dbsync the responsible of managing the DB.
+ * @param storage storage 1 Store database in memory, disk otherwise.
+ * @param sync_interval Interval when the synchronization will be performed.
+ * @param file_limit Maximum number of files to be monitored
+ * @param sync_callback Callback to send the synchronization messages.
+ * @param log_callback Callback to perform logging operations.
  */
-fim_registry_value_data* fim_db_decode_registry_value(sqlite3_stmt* stmt);
-
+enum FIMDBErrorCodes fim_db_init(int storage, int sync_interval, int file_limit, fim_sync_callback_t sync_callback,
```

Comment:
Avoid having the parameter list in the same line when more than two or three. Let's define as follows:
```
enum FIMDBErrorCodes fim_db_init(int storage,
                                 int sync_interval,
                                 int file_limit,
                                 fim_sync_callback_t sync_callback,
                                 logging_callback_t log_callback);
```

Apply this to all the functions in this PR please.